addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -78,29 +78,17 @@ function factory(dependencies) {
          * @param {function} res
          */
         async handleAddChannelAutocompleteSource(req, res) {
-            const value = req.term;
-            const escapedValue = owl.utils.escape(value);
-            this.update({ addingChannelValue: value });
-            const domain = [
-                ['channel_type', '=', 'channel'],
-                ['name', 'ilike', value],
-            ];
-            const fields = ['channel_type', 'name', 'public', 'uuid'];
-            const result = await this.async(() => this.env.services.rpc({
-                model: "mail.channel",
-                method: "search_read",
-                kwargs: {
-                    domain,
-                    fields,
-                },
-            }));
-            const items = result.map(data => {
-                let escapedName = owl.utils.escape(data.name);
-                return Object.assign(data, {
+            const threads = await this.env.models['mail.thread'].searchChannelsToOpen({ keyword: req.term, limit: 10 })
+            const items = threads.map((thread) => {
+                const escapedName = owl.utils.escape(thread.name);
+                return {
+                    channel_type: thread.channel_type,
                     label: escapedName,
-                    value: escapedName
-                });
+                    name: thread.name,
+                    value: escapedName,
```

Comment:
`id` is missing from here, selecting a channel that is already joined in discuss sidebar leads to a crash

Reply:
I'm thinking apart from `label` and `value` which are used by the lib, `id` might be the only necessary field, not `channel_type` and `name` (if referring to usage of `ui.item`).

Reply:
I just adapt the code to have the same object.

Reply:
I will remove `channel_type` and `name`